deltacat/compute/compactor/steps/dedupe.py

```diff
@@ -64,13 +65,15 @@ def union_primary_key_indices(
                                  f"delta stream position found in round "
                                  f"completion info: {round_completion_info}")
             prior_pk_index_table = pa.concat_tables(tables)
-            prior_pk_index_table = sc.append_stream_position_column(
-                prior_pk_index_table,
-                repeat(
-                    prev_compacted_delta_stream_pos,
-                    len(prior_pk_index_table),
-                ),
-            )
+            if not sc.if_stream_position_column_exists(prior_pk_index_table):
+                logger.info(f"peach: append stream position column when pk: {build_initial_primary_key_indices}")
```

Comment:
might need better logger info to make it more readable?

Reply:
Nice catch! Sorry I left the debug lines here